cartography/intel/semgrep/findings.py

```diff
@@ -0,0 +1,217 @@
+            response.raise_for_status()
+            if prev_request == response.request.url:
+                logger.warning(
+                    "Duplicate request detected. Breaking the loop to avoid infinite requests.",
+                )
+                break
+            prev_request = response.request.url
+            if response.status_code == 200:
+                data = response.json()
+                vulns = data["vulns"]
+                cursor = data.get("cursor")
+                has_more = data.get("hasMore", False)
+                all_vulns.extend(vulns)
+        except requests.RequestException as e:
+            logger.error(
+                "Could not complete request to the SCA Semgrep API: %s", e, exc_info=e,
+            )
+            raise e
+        except Exception as e:
+            logger.error("Erorr retrieving Semgrep SCA vulns info: %s", e, exc_info=e)
+            raise e
+    return all_vulns
+
+
+def transform_sca_vulns(raw_vulns: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+    """
+    Transforms the raw SCA vulns response from Semgrep API into a list of dicts
+    that can be used to create the SemgrepSCAFinding nodes.
+    """
+    vulns = []
+    for vuln in raw_vulns:
+        sca_vuln: Dict[str, Any] = {}
+        # Mandatory fields
+        unique_id = f"{vuln['repositoryName']}|{vuln['advisory']['ruleId']}"
+        sca_vuln["id"] = unique_id
+        sca_vuln["repositoryName"] = vuln["repositoryName"]
+        sca_vuln["ruleId"] = vuln["advisory"]["ruleId"]
+        sca_vuln["title"] = vuln["advisory"]["title"]
+        sca_vuln["description"] = vuln["advisory"]["description"]
+        sca_vuln["ecosystem"] = vuln["advisory"]["ecosystem"]
+        sca_vuln["severity"] = vuln["advisory"]["severity"]
+        sca_vuln["reachability"] = vuln["advisory"]["reachability"]
+        sca_vuln["reachableIf"] = vuln["advisory"]["reachableIf"]
+        sca_vuln["exposureType"] = vuln["exposureType"]
+        dependency = f"{vuln['matchedDependency']['name']}|{vuln['matchedDependency']['versionSpecifier']}"
+        sca_vuln["matchedDependency"] = dependency
+        sca_vuln["dependencyFileLocation_path"] = vuln["dependencyFileLocation"]["path"]
+        sca_vuln["dependencyFileLocation_url"] = vuln["dependencyFileLocation"]["url"]
+        # Optional fields
+        cves = vuln.get("advisory", {}).get("references", {}).get("cveIds")
+        if len(cves) > 0:
+            # Take the first CVE
+            sca_vuln["cveId"] = vuln["advisory"]["references"]["cveIds"][0]
+        if vuln.get('closestSafeDependency'):
+            dep_fix = f"{vuln['closestSafeDependency']['name']}|{vuln['closestSafeDependency']['versionSpecifier']}"
+        else:
+            dep_fix = None
+        sca_vuln["closestSafeDependency"] = dep_fix
+        ref_urls = vuln["advisory"].get("references", {}).get("urls", [])
+        if ref_urls:
+            sca_vuln["ref_urls"] = ",".join(ref_urls)
```

Comment:
non-blocker: this node property's value can also be an array.

Reply:
Does the [querybuilder](https://github.com/lyft/cartography/blob/f657a0f73c0cc449a27d8a490f2b86d66d31951d/cartography/graph/querybuilder.py) supports passing arrays?

Reply:
Wow, first time that I will see list as node properties, will give it a try!

Reply:
I forget what this conversation thread was originally about since GitHub's commenting seems to have moved it to a different line.

In general though let's try to use lists in Neo4j nodes as little as possible since it's easier to test for relationship connections. Using lists does make sense though for things where it's too heavyweight to create a whole new node type.